### PR TITLE
Add MapboxNavigationNative

### PR DESCRIPTION
**Pull request by frederoni**
This PR pulls in MapboxNavigationNative as a dependency.

note that this PR is targeting the `navigation-native` branch.

@bsudekum @akitchen

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 binary "https://www.mapbox.com/ios-sdk/Mapbox-iOS-SDK.json" ~> 4.0
-github "mapbox/MapboxDirections.swift" ~> 0.21.0
+binary "https://s3.amazonaws.com/mapbox/mapbox-navigation-native/ios/builds/Mapbox-iOS-Navigation-Native.json"
+#github "mapbox/MapboxDirections.swift" ~> 0.21.0
```
Comment (bsudekum) [inline Cartfile]: It's fine to just uncomment this and remove it.

**Comment by akitchen** [comment]
@frederoni et. al -- what do you suggest we do about the Cocopods problem with this one?

**Comment by frederoni** [comment]
@akitchen should be able to work around it with some tweaks to the podspec or Podfile. If not, we could always add a private pod repo for the time being. The issue here is that a podspec can only depend on a published pod.

**Comment by frederoni** [comment]
Sure, let's consider the unpublished pod as tail work.

**Pull request merged**